### ISSUE TITLE
usage Client Credentials Authentication

### ISSUE THREAD

**Issue by pka420** [globus/globus-sdk-python#759]
Hi, 

i created a new web app on globus, created `client_id` and `client_secret`.
When I authenticate using `ConfidentialAppAuthClient` [here](https://globus-sdk-python.readthedocs.io/en/stable/examples/client_credentials.html), I am able to obtain tokens.

But, the tranfer_token is not authorized to be used with any of my endpoints. 

The token belongs to `client_id@clients.auth.globus.org` but the endpoints belong to a user.

How can I use ConfidentialAppAuthClient to get authorization for changing my endpoints?

**Comment by sirosen** [comment]
Hi there! I'm not sure that this is really a globus-sdk question per-se, so you may get better support via support@globus.org

That said, all you should need to do to use client credentials to manage an endpoint is add the client to the administrators on that endpoint.
The simplest way would be via the web GUI, where you can navigate to the Roles tab and add an identity there.
Here's a sample screenshot from one of my own endpoints:
![image](https://github.com/globus/globus-sdk-python/assets/1300022/47e16478-238e-496b-b4e8-b670f9117804)

Just select the Assign New Role option on the right and plug in your client ID as an Administrator.

You can also do this via the globus-cli ([install doc](https://docs.globus.org/cli/#installation)) with
```sh
$ globus endpoint role create $ENDPOINT_ID --identity $CLIENT_ID --role administrator
```

Once your client is an admin of the endpoint, it should have full management capabilities just like your user account.

Notably, some role assignment features require a subscription, so this might not work for you if you don't have a subscription. (I'm not an expert on the full suite of restrictions, but I believe that role assignment is unavailable on GCP without a subscription, but that it is available on GCS.)
If you can't use a role to delegate access, then it won't be possible for your client to manage endpoints belonging to other users.

**Comment by pka420** [comment]
Hi @sirosen 

I asked the admin and we tried to do the same thing again.

Here is what we tried:

1. created an app by going to settings -> developers. 
![image](https://github.com/globus/globus-sdk-python/assets/30459586/fdb3784a-6c67-4e01-9603-645ee888942c)

2. We copied the client_id it generated and tried to assign it the role of administrator 
![image](https://github.com/globus/globus-sdk-python/assets/30459586/b304c108-fe2b-4300-9368-ee4aea23a15b)

But the client_id does not show up, are we missing something here? 

The globus installation on the server is GCS5.
